os/drivers/input/ist415.c

```diff
@@ -666,7 +692,10 @@ static void ist415_lockup_work(struct ist415_dev_s *dev)
 		ist415_disable(dev);
 		ist415_forced_release(dev);
 		ist415_reset(dev, false);
-		if (dev->suspend) {
+
+		if (dev->forcedoff) {
+			ist415_power_off(dev);
+		} else if (dev->suspend) {
```

Comment:
if both of forceoff & suspend never set as true, I think you can define enum regarding state and use dev->state instead of these 3 states.